packages/core/src/utils/__tests__/security.spec.ts

```diff
@@ -0,0 +1,279 @@
+          },
+        ]);
+      });
+    });
+
+    describe('invalid - no authorization header', () => {
+      test('returns an error object with 401 code', () => {
+        expect(
+          validateSecurity<any, any>({ headers: {} }, { security: [{ scheme: 'bearer', type: 'http' }] }),
+        ).toStrictEqual([
+          {
+            headers: {
+              'WWW-Authenticate': 'Bearer',
+            },
+            message: 'Invalid security scheme used',
+            name: 'Unauthorised',
+            status: 401,
+          },
+        ]);
+      });
+    });
+  });
+
+  describe('oauth2', () => {
+    describe('valid', () => {
+      test('passes when a bearer token is provided', () => {
+        expect(
+          validateSecurity<any, any>(
+            { headers: { authorization: 'Bearer abc123' } },
+            { security: [{ type: 'oauth2' }] },
+          ),
+        ).toStrictEqual([]);
+      });
+    });
+
+    describe('invalid', () => {
+      test('returns an error object with 401 code', () => {
+        expect(
+          validateSecurity<any, any>(
+            { headers: { authorization: 'Digest abc123' } },
+            { security: [{ type: 'oauth2' }] },
+          ),
+        ).toStrictEqual([
+          {
+            headers: {
+              'WWW-Authenticate': 'OAuth2',
+            },
+            message: 'Invalid security scheme used',
+            name: 'Unauthorised',
+            status: 401,
+          },
+        ]);
+      });
+    });
+  });
+
+  describe('openid', () => {
+    describe('valid', () => {
+      test('passes when a bearer token is provided', () => {
+        expect(
+          validateSecurity<any, any>(
+            { headers: { authorization: 'Bearer abc123' } },
+            { security: [{ type: 'openIdConnect' }] },
+          ),
+        ).toStrictEqual([]);
+      });
+    });
+
+    describe('invalid', () => {
+      test('returns an error object with 401 code', () => {
+        expect(
+          validateSecurity<any, any>(
+            { headers: { authorization: 'Digest abc123' } },
+            { security: [{ type: 'openIdConnect' }] },
+          ),
+        ).toStrictEqual([
+          {
+            headers: {
+              'WWW-Authenticate': 'OpenID',
+            },
+            message: 'Invalid security scheme used',
+            name: 'Unauthorised',
+            status: 401,
+          },
+        ]);
+      });
+    });
+  });
+
+  describe('API KEY', () => {
+    describe('in header', () => {
+      describe('valid', () => {
+        test('passes when the correct header is provided', () => {
+          expect(
+            validateSecurity<any, any>(
+              { headers: { 'x-api-key': 'abc123' } },
+              { security: [{ in: 'header', type: 'apiKey', name: 'x-api-key' }] },
+            ),
+          ).toStrictEqual([]);
+        });
+      });
+
+      describe('invalid', () => {
+        test('returns an error object with 401 code', () => {
```

Comment:
@philsturgeon, there's a couple of `test('returns an error object with 401 code` there. Would you rather see them all changed to `it('fails with XYZ`?

Reply:
Yeah generally we use the `it` stuff because we are following describe-it, so might as well. 

Check this awesome site out http://www.betterspecs.org/ I was a big fan when I was over in ruby land, and we recommended it for everyone trying to figure testing naming and practices. It doesnt all apply because ruby, but things like `let(:foo)` is just `const foo` and a lot of it does work. 

Using more `describe` (because we dont have `context`) for setup and description of the state, then the it just describes the outline of the test without getting too stuck into exactly what assertions are.

Reply:
@philsturgeon, the tests are now adjusted, hopefully adjusted correctly ;)